src/events/events.gateway.ts

```diff
@@ -61,7 +67,14 @@ export class EventGateway {
       }
       await this.logService.create(log);
       socket.disconnect();
-    }
+      delete this.socketIdMap[`${this.keyOfMap}`];
+
+    }      
+  }
+
+  handleDisconnect(socket: Socket){
+    delete this.socketIdMap[`${this.keyOfMap}`];
```

Comment:
this seems a bit overcomplicated. Can you just iterate through object vals, check if id of the socket matches the value, delete by key.
Also, is this class instantiated for each connection? 'cause if not - you create a race of condition, always deleting last connected user.